enrichments/trace/internal/elastic/span.go

```diff
@@ -417,15 +422,38 @@ func (s *spanEnrichmentContext) setDestinationService(span ptrace.Span) {
 }
 
 type spanEventEnrichmentContext struct {
-	exception bool
+	exception        bool
+	exceptionEscaped bool
+
+	exceptionType       string
+	exceptionMessage    string
+	exceptionStacktrace string
 }
 
 func (s *spanEventEnrichmentContext) enrich(
 	se ptrace.SpanEvent,
 	cfg config.SpanEventConfig,
 ) {
+	// TODO(lahsivjar): Is error/log context handling be done in es-exporter?
+	// Ref: https://github.com/elastic/apm-data/blob/59d33b8113d629f43dc82f9b3931ceda63a19a7a/input/otlp/traces.go#L1243-L1260
+
 	// Extract top level span event information.
 	s.exception = se.Name() == "exception"
+	if s.exception {
+		se.Attributes().Range(func(k string, v pcommon.Value) bool {
+			switch k {
+			case semconv.AttributeExceptionEscaped:
+				s.exceptionEscaped = v.Bool()
+			case semconv.AttributeExceptionType:
+				s.exceptionType = v.Str()
+			case semconv.AttributeExceptionMessage:
+				s.exceptionMessage = v.Str()
+			case semconv.AttributeExceptionStacktrace:
+				s.exceptionStacktrace = v.Str()
```

Comment:
@gregkalapos These attributes are mostly 1<>1 mappings however, I didn't see them in the otel-data plugin so added it here. Do you think we should do this here or in the otel-data plugin?

Reply:
Yeah, aliases are how we have been handling such fields. Will wait for @gregkalapos to confirm

Reply:
Yes, I'd be also for aliases as well.

One exception (no pun intended) may be the `handled` flag: from what I remember from the UI testing, the UI access that field, so it always needs to be present (We could question why, but from what I remember, that's how it is today), but [`exception.escaped`](https://opentelemetry.io/docs/specs/semconv/exceptions/exceptions-spans/#attributes) is only recommended. So we always need to populate the `handled` flag.

That's the only edge case which comes to my mind, I'd expect the other ones to work with aliases.

Reply:
Sounds good. Could you add these aliases, Greg? One challenge will be how to add the attribute aliases just for logs, without these getting overridden with the standard set of attribute aliases. Maybe we'll need to add them to the [alias definition](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/otel-data/src/main/resources/component-templates/semconv-resource-to-ecs%40mappings.yaml) that applies to all signals.
Also, I'm not sure if we need an alias for the exception stacktrace, since we likely need to fetch that from _source anyway, because IINM, we don't have a stored field for it anyway?

Reply:
But we don't even need the aliases, don't we? 🤔 This should all work out via `passthrough`. 

Following could be sent via attributes and would be visible top level:
- `exception.type`
- `exception.message`
( above 2 are in SemConv - I don't think we even need to do anything with those here in the enrichment lib)
- `exception.handled`: that is not part of OTel, se we need to set it here.

I opened a draft: https://github.com/elastic/elasticsearch/pull/113636 I commented out all the aliases, the test is still green which queries `exception.type` top level.

Reply:
@gregkalapos I think the alias should be for `error.exception.{type, message}` -> `exception.{type, message}`

Reply:
Also, I think we might need to put alias for stacktrace too: `error.stack_trace` -> `exception.stacktrace`

Reply:
Oh, [I see](https://github.com/elastic/kibana/blob/2ec24f2f48b67a1c928fa6edb84214de57b59dd2/packages/kbn-apm-types/src/es_fields/apm.ts#L108-L111).

Ok, we can do `error.exception.{type, message}` -> `exception.{type, message}`; I'll update the PR.

Same for stack trace.